packages/expo-updates/ios/EXUpdates/Database/EXUpdatesDatabase.m

```diff
@@ -333,9 +365,13 @@ - (void)deleteUpdates:(NSArray<EXUpdatesUpdate *> *)updates error:(NSError ** _N
 
 - (nullable NSArray<EXUpdatesUpdate *> *)launchableUpdatesWithConfig:(EXUpdatesConfig *)config error:(NSError ** _Nullable)error
 {
+  // if an update has successfully launched at least once, we treat it as launchable
+  // even if it has also failed to launch at least once
   NSString *sql = [NSString stringWithFormat:@"SELECT *\
   FROM updates\
   WHERE scope_key = ?1\
+  AND is_outdated = 0\
+  AND (successful_launch_count > 0 OR failed_launch_count < 1)\
```

Comment:
An update can inconsistently succeed or fail e.g. it could succeed the first time and fail the second (successes = 1, failures = 1). This check is to see if the update has ever been successful or if it has never run at all. Another way to read this check is "successful_launch_count > 0 PR successful_launch_count + failed_launch_count = 0`.